src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1004,7 +1008,16 @@ public void preloadInstance(TreeElement node) {
 
     public boolean postProcessInstance() {
         actionController.triggerActionsFromEvent(Actions.EVENT_XFORMS_REVALIDATE, elementsWithActionTriggeredByToplevelEvent, this);
-        return postProcessInstance(mainInstance.getRoot());
+        boolean instanceChanged = postProcessInstance(mainInstance.getRoot());
+
+        List<Pair<String, String>> fields = saveTos.stream().map(saveTo -> {
+            IDataReference reference = saveTo.getFirst();
+            String answer = mainInstance.resolveReference(reference).getValue().getDisplayText();
```

Comment:
I think we actually want the underlying value not the display text. This makes a different for selects.

This won't be great performance-wise. Probably fine to start here but I think eventually we'll want to do a single parse pass to get everything out. Might not be the first part of instance save we want to optimize, though, it's pretty slow already.

Reply:
Do you have a scenario we could use for a failing test for this? I've tried writing one with selects (`fillingFormWithSelectSaveTo_andWithCreate_savesValuesCorrectlyToEntity`), but it's happy.